src/odemis/util/driver.py

```diff
@@ -157,12 +160,55 @@ def guessActuatorMoveDuration(actuator, axis, distance, accel=DEFAULT_ACCELERATI
     if not axis in actuator.axes:
         raise KeyError("The actuator component %s is expected to have %s axis, but it does not." % (actuator, axis))
     if model.hasVA(actuator, "speed"):
-        speed = actuator.speed.value.get("z", None)
+        speed = actuator.speed.value.get(axis, None)
     if speed is None:
         speed = DEFAULT_SPEED  
     return estimateMoveDuration(distance, speed, accel) 
 
 
+class ProgressiveMove(model.ProgressiveFuture):
+    def __init__(self, comp, pos):
+        self.name = comp.name
+        est_start = time.time() + 0.1
+
+        # calculate the time the mirror needs to move from the current position to the requested position
+        current_pos = comp.position.value
+        current_speed = comp.speed.value
+        axes_total_move_time = 0.0
+
+        for ax in pos.keys():
+            if ax in current_speed.keys():
+                axes_total_move_time += abs((current_pos[ax] - pos[ax]) / current_speed[ax])
```

Comment:
Using the some is in most cases a very pessimistic estimation. In practice, if multiple axes move, most actuators will move all of them at the same time. So you can just use the `max`. It should give a more realistic time estimation.

Reply:
Good suggestion, I used a slightly different approach hopefully you'll agree with that as well.